Formula/d/dmd.rb

```diff
@@ -65,9 +65,19 @@ def install
     cp_r ["phobos/std", "phobos/etc"], include/"dlang/dmd"
     lib.install Dir["druntime/**/libdruntime.*", "phobos/**/libphobos2.*"]
 
+    dflags = "-I#{opt_include}/dlang/dmd -L-L#{opt_lib}"
+    # We include the -ld_classic linker argument in dmd.conf because it seems to need
+    # changes upstream to support the newer linker:
+    # https://forum.dlang.org/thread/jwmpdecwyazcrxphttoy@forum.dlang.org?page=1
+    # https://github.com/ldc-developers/ldc/issues/4501
+    #
+    # Also, macOS can't run CLT/Xcode new enough to need this flag, so restrict to Ventura
+    # and above.
+    dflags << " -L-ld_classic" if OS.mac? && MacOS.version > :monterey
```

Comment:
I think this needs to switch on XCode 15 or later since that's what shipped the new linker

Reply:
I think I agree now that switching on Xcode 15 ( logical choice for how to configure it. Originally I was thinking that this config in `dmd.conf` could be problematic for a user if the wasn't present and they _hadn't_ installed Xcode 15 yet, and then installed it later. But that seems like it's increasingly becoming an edgecase. Plus people can just edit the file themselves.